tests/test_tracer.py

```diff
@@ -32,3 +35,51 @@ def test_init_openai_only():
     )
     assert len(anthropic_response.content) > 0
     assert not hasattr(anthropic_response, "impacts")
+
+    EcoLogits.init(providers=["anthropic"]) # adds anthropic
+
+    openai_response = openai_client.chat.completions.create(
+        model="gpt-3.5-turbo",
+        messages=[{"role": "user", "content": "Hello World!"}]
+    )
+    assert len(openai_response.choices) > 0
+    assert hasattr(openai_response, "impacts")
+
+    anthropic_response = anthropic_client.messages.create(
+        max_tokens=100,
+        messages=[{"role": "user", "content": "Hello World!"}],
+        model="claude-3-5-sonnet-20240620",
+    )
+    assert len(anthropic_response.content) > 0
+    assert hasattr(anthropic_response, "impacts")
+
+@pytest.mark.vcr
+def test_init_with_different_mixes():
+    client = Anthropic()
+
+    EcoLogits.init(electricity_mix_zone="FRA")
+    message_under_french_mix = client.messages.create(
+        max_tokens=1024,
+        messages=[
+            {
+                "role": "user",
+                "content": "Hello, Claude",
+            }
+        ],
+        model="claude-3-opus-20240229",
+    )
+
+    EcoLogits.init(electricity_mix_zone="BEL")
+    message_under_belgian_mix = client.messages.create(
+        max_tokens=1024,
+        messages=[
+            {
+                "role": "user",
+                "content": "Hello, Claude",
+            }
+        ],
+        model="claude-3-opus-20240229",
+    )
+
+    assert message_under_belgian_mix.impacts.energy.value.min != message_under_french_mix.impacts.energy.value.min
+    
```

Comment:
Some remarks on this test:

1. I would use OpenAI with [deterministic output](https://cookbook.openai.com/examples/reproducible_outputs_with_the_seed_parameter) (i.e. the seed set to `0`). This will make sure we have the same output for the same input parameters, and thus make the two request impacts comparable.
2. As it is written, the test should be failing. Changing the electricity mix does not impact the energy consumption. We should test for GWP impacts instead. The test is currently not failing because we compare equality of floats that are not exactly the same 0.008138263895 ≈ 0.0081090964725
3. I would also change the comparison to FRA < WOR impacts instead of just checking inequality.

Reply:
Completely agree! Went too fast on this one